src/QMCHamiltonians/CoulombPBCAA.cpp

```diff
@@ -216,6 +216,10 @@ CoulombPBCAA::Return_t CoulombPBCAA::evaluate_sp(ParticleSet& P)
     }
     else
     {
+      if (!P.getPerParticleSKState())
+      {
+        P.turnOnPerParticleSK();
```

Comment:
I doubt this is the right place to flip the switch which doesn't make the data ready which seems needed by the evalluate below.

Reply:
It looks like `StructFact::turnOnStorePerParticle` calls `FillRhok`, which initializes the `eikr_r` array.  I'm open to suggestions for a better way or place to enable it.

Reply:
I feel like this belongs in the constructor, its called on the ref ParticleSet passed there if ComputeForces == true. This seems a bit wrong as well but we can unravel one level of tangle at a time. I would assume that the evaluate_sp ParticleSet& should be getting called on the ref passed to the constructor or a clone of it. No one would expect to have to manipulate the state of the ParticleSet here to make it valid.

Reply:
I'm not sure it's easily possible to move this to the constructor - this is not know at construction time.  I can try move it up the call stack so it's in a more general place.